pymeasure/instruments/hp/hp8753e.py

```diff
@@ -0,0 +1,472 @@
+        """Get the active Scattering or Measuring Parameter being measured.
+        (str from ['S11', 'S12', 'S21', 'S22', 'A/R', 'B/R', 'A/B', 'A', 'B', 'R'])"""
+        for key in self.MEASURING_PARAMETER_MAP:
+            if int(self.ask(f"{self.MEASURING_PARAMETER_MAP[key]}?")) == 1:
+                return key
+        return None
+
+    @measuring_parameter.setter
+    def measuring_parameter(self, value):
+        """Set the active Measuring Parameter to be measured.
+        (str from ['S11', 'S12', 'S21', 'S22', 'A/R', 'B/R', 'A/B', 'A', 'B', 'R'])"""
+        if value in self.MEASURING_PARAMETER_MAP:
+            self.write(self.MEASURING_PARAMETER_MAP[value])
+        else:
+            raise ValueError(
+                f"Invalid value '{value}' scattering parameter requested for \
+                {self._manu} {self._model}. Valid values are: {self.MEASURING_PARAMETER_MAP}"
+            )
+
+    IFBW = Instrument.control(
+        "IFBW?",
+        "IFBW%d",
+        f"""Control the IF Bandwidth of the instrument for a scan sweep.
+        (int from the set {ALLOWED_BANDWIDTH}).""",
+        cast=lambda x: int(float(x)),
+        validator=strict_discrete_set,
+        values=ALLOWED_BANDWIDTH,
+    )
+
+    def reset(self):
+        """Reset the instrument. May cause RF Output power to be enabled!"""
+        self.write("*RST")
+        sleep(0.25)
+
+    def scan(self, timeout=10):
+        """Initiates a scan with the number of averages specified and
+        blocks until the operation is complete.
+
+        :param timeout: Optional value in seconds to add to the sweep time before timeout occurs.
+        :raise TimeoutError: If the VNA fails to complete the sweep before timeout occurs.
+        :return: None.
+        """
+
+        # get time to perform sweep
+        sweep_time = self.sweep_time + timeout
+
+        # get number of averages if enabled
+        if self.averaging_enabled:
+            sweep_time = sweep_time * self.averages
+
+        self.scan_single()
+
+        # create a time limit
+        start = now()
+
+        # All queries will block until the scan is done, so use NOOP? to check.
+        # These queries will time out after several seconds though,
+        # so query repeatedly until the scan finishes.
+        while True:
+            try:
+                # status = self.ask("NOOP?")
+                self.ask("NOOP?")
+                break
+            except VisaIOError:
+                pass
+
+            # calculate time sweep should be complete by
+            if now() > (start + sweep_time):
+                raise TimeoutError(
+                    f"VNA Scan took longer than {sweep_time} seconds to complete and timed out."
+                )
+
+        sleep(0.5)
+        while self.adapter.connection.bytes_in_buffer > 0:
+            _ = self.read()
+            sleep(0.25)
+
+    def scan_single(self):
+        """Initiates a single scan or N scans averaged based on averaging
```

Comment:
Docstrings should also end with a period:
```suggestion
        """Initiate a single scan or N scans averaged based on averaging.
```